staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation_test.go

```diff
@@ -2426,6 +2426,91 @@ func TestMessageExpression(t *testing.T) {
 	}
 }
 
+func TestReasonAndFldPath(t *testing.T) {
+	forbiddenReason := func() *field.ErrorType {
+		r := field.ErrorTypeForbidden
+		return &r
+	}()
+	tests := []struct {
+		name      string
+		schema    *schema.Structural
+		obj       interface{}
+		errors    []string
+		errorType field.ErrorType
+	}{
+		{name: "Return error based on input reason",
+			obj: map[string]interface{}{
+				"f": map[string]interface{}{
+					"m": 1,
+				},
+			},
+			schema: withRulePtr(objectTypePtr(map[string]schema.Structural{
+				"f": withReasonAndFldPath(objectType(map[string]schema.Structural{"m": integerType}), "self.m == 2", "", forbiddenReason),
+			}), "1 == 1"),
+			errorType: field.ErrorTypeForbidden,
+			errors:    []string{"root.f: Forbidden"},
+		},
+		{name: "Return error default is invalid",
+			obj: map[string]interface{}{
+				"f": map[string]interface{}{
+					"m": 1,
+				},
+			},
+			schema: withRulePtr(objectTypePtr(map[string]schema.Structural{
+				"f": withReasonAndFldPath(objectType(map[string]schema.Structural{"m": integerType}), "self.m == 2", "", nil),
+			}), "1 == 1"),
+			errorType: field.ErrorTypeInvalid,
+			errors:    []string{"root.f: Invalid"},
+		},
+		{name: "Return error based on input fieldPath",
+			obj: map[string]interface{}{
+				"f": map[string]interface{}{
+					"m": 1,
+				},
+			},
+			schema: withRulePtr(objectTypePtr(map[string]schema.Structural{
+				"f": withReasonAndFldPath(objectType(map[string]schema.Structural{"m": integerType}), "self.m == 2", ".m", forbiddenReason),
```

Comment:
Also test:

- Access to field nested more than one level deep
- Access to a list item (if we're supporting that like the API documentation suggests)
- Access to a map entry (if we're supporting that)
- An invalid json path (need multiple cases here for things like missing prefix `.`, illegal chars, `[` without a following `]`, ...) - This is going to be tough to get right without a grammer. https://pkg.go.dev/text/scanner might help, it can be used to establish a grammer in very few lines of code (example: https://github.com/kubernetes/kube-openapi/pull/381/files#diff-8785f6b3fa88f7876786b2108a445b59075ddfecb6b923ffc8ea1d7469892c86R29)
- A valid json path to a field that does not exist

Reply:
Added tests